feature/amazon/src/test/java/com/revenuecat/purchases/amazon/AmazonCacheTest.kt

```diff
@@ -0,0 +1,157 @@
+
+        val receiptTermSkus = underTest.getReceiptSkus()
+
+        assertThat(receiptTermSkus).isEqualTo(
+            mapOf(
+                "1234abcdreceiptid" to "com.revenuecat.subscription.weekly",
+                "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+            )
+        )
+    }
+
+    @Test
+    fun `set receipt term skus on an empty cache`() {
+        val expected = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly",
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(expected)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
+
+        expected.forEach { (key, value) ->
+            assertThat(actualStoredMapAsJSON[key]).isEqualTo(value)
+        }
+    }
+
+    @Test
+    fun `set receipt term skus on a non empty cache`() {
+        val alreadyCached = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly"
+        )
+
+        val cachedReceiptsToTermSkusJSON = getStoredJSONFromMap(alreadyCached)
+        cache.stubCache[underTest.amazonPostedTokensKey] = cachedReceiptsToTermSkusJSON
+
+        val newToCache = mapOf(
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(newToCache)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+
+        val expected = alreadyCached + newToCache
+
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
+
+        expected.forEach { (key, value) ->
+            assertThat(actualStoredMapAsJSON[key]).isEqualTo(value)
+        }
+    }
+
+    @Test
+    fun `overriding a receipt term sku`() {
+        val alreadyCached = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly"
+        )
+
+        val cachedReceiptsToTermSkusJSON = getStoredJSONFromMap(alreadyCached)
+        cache.stubCache[underTest.amazonPostedTokensKey] = cachedReceiptsToTermSkusJSON
+
+        val expected = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(expected)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
```

Comment:
same thought regarding getting the size in other ways. there's also `keySet()` if `length()` doesn't work, so we get a Set instead of an iterator and can get the size directly without doing `asSequence`